examples/communication/ip/KnockKnockSample/src/com/digi/xbee/api/knockknock/WebServer.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (c) 2016 Digi International Inc.,
+ * All rights not expressly granted are reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * Digi International Inc. 11001 Bren Road East, Minnetonka, MN 55343
+ * =======================================================================
+ */
+package com.digi.xbee.api.knockknock;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.net.ServerSocket;
+import java.net.Socket;
+
+/**
+ * Simple web server that responds to clients with a text message.
+ */
+public class WebServer {
+
+	/**
+	 * Starts the web server in the given port.
+	 *
+	 * @param port The port number of the web server.
+	 */
+	public static void start(final int port) {
+		new Thread(new Runnable() {
+			@Override
+			public void run() {
+				ServerSocket serverSocket = null;
+				Socket client = null;
+				try {
+					serverSocket = new ServerSocket(port);
+					while (true) {
+						client = serverSocket.accept();
+						PrintWriter out = new PrintWriter(client.getOutputStream());
+						BufferedReader in = new BufferedReader(new InputStreamReader(client.getInputStream()));
+						KnockKnockProtocol kkp = new KnockKnockProtocol();
```

Comment:
KnockKnockProtocol could be instantiated outside the while loop. It is constant for any socket client.